src/shared/auth/jwt-payload.interface.ts

```diff
@@ -1,9 +1,16 @@
 import { Blockchain } from 'src/blockchain/shared/enums/blockchain.enum';
 import { UserRole } from './user-role.enum';
 
-export interface JwtPayload {
+export interface JwtBasicPayload {
   id: number;
   address: string;
   role: UserRole;
+}
+
+export interface JwtPayload extends JwtBasicPayload {
   blockchains: Blockchain[];
 }
+
+export interface JwtCompanyPayload extends JwtBasicPayload {
+  isKycClient: boolean;
```

Comment:
I would solve this with different roles, e.g. `Wallet` and `KycWallet` or similar.